src/mlte/spec/condition.py

```diff
@@ -0,0 +1,108 @@
+"""
+Defines a condition for a measurement to be approved, including the validator to use and the needed threshold.
+"""
+
+from __future__ import annotations
+
+from typing import Any
+
+from mlte.validation import Result
+from mlte.value import Value
+
+# -----------------------------------------------------------------------------
+# Condition
+# -----------------------------------------------------------------------------
+
+
+class Condition:
```

Comment:
So I get this can be confusing, mainly because this PR is a "part 1" for the declarative spec, and won't be fully completed until the next PR. When I initially came up with the Condition class, it was an association of something you wanted to validate eventually, which included a measurement type, a validation method and arguments for that. In our last meeting, you suggested, as you are indicating above, to basically rename the Validator into Condition, and use some other names for this association (I think RegExp), and changing it to be a thunk. Don't worry, I got that. But as I mentioned in the PR, all of that is not included yet in this PR. I am still using my initial definition of Condition, which is not the condition you want, for this first step. I will rename the Validator to Condition and work with the thunks and everything in the next PR. Since there was so much renaming going on, and I already had a Condition defined in my old way here, I prefered to leave that to the second PR. We can discuss this at our upcoming meeting, and I understand if you would prefer for me to continue working on this PR instead of dividing it into two steps, if you are not confortable with this as an intermediate step.

Reply:
Resolved during discussion.